parser/parser/include/ruby_parser/lexer.hh

```diff
@@ -190,13 +190,13 @@ public:
     // Main interface consumed by yylex function in parser
     token_t advance();
 
-    // Useful for error recovery. Manually pushes `tok_to_push` (usually: the parser's current
-    // lookahead token) back onto the front of the token_queue, and then allocates a new token and
-    // returns it.
-    token_t unadvance(token_t tok_to_push, token_type type, size_t start, size_t end, const std::string &str);
-
     std::string_view tok_view_from_offsets(size_t start, size_t end) const;
 
+    // Useful for error recovery. Manually sets `_p` so that the next call to `advance` starts
+    // from the desired location. If you're using this, you likely want to set the lexer state
+    // menually with one of the below helpers.
+    void rewind(size_t newPos);
```

Comment:
Similar comment here: unless we're sure that calling one of the below helpers is sufficient to fully roll back lexer state (and I don't think we are sure of that), it would be good to note that here.